browser/ui/webui/brave_shields/shields_data_handler.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <utility>
+
+#include "brave/browser/ui/webui/brave_shields/shields_data_handler.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "ui/webui/mojo_bubble_web_ui_controller.h"
+
+ShieldsDataHandler::ShieldsDataHandler(
+    mojo::PendingReceiver<brave_shields_panel::mojom::DataHandler>
+        data_handler_receiver,
+    ui::MojoBubbleWebUIController* webui_controller)
+    : data_handler_receiver_(this, std::move(data_handler_receiver)),
+      webui_controller_(webui_controller) {
+  auto* profile = Profile::FromWebUI(webui_controller_->web_ui());
+  DCHECK(profile);
+  Browser* browser = chrome::FindLastActiveWithProfile(profile);
+  browser->tab_strip_model()->AddObserver(this);
```

Comment:
This is not the right way to get the current window, you should be getting the webcontents from the webui and calling `FindBrowserWithWebContents`

Reply:
this is bubble's web contents